### PR TITLE
Add check uncheck bulk checkboxes to role editor

### LINKED ISSUE #11: Add Bulk Select/Deselect Checkbox(s)
Hy,
Can you add a mass selection function for member roles by categories.

### PR DESCRIPTION
**Pull request by ThemeGravity**
Resolves #11

### CODE REVIEW

**Comment by ThemeGravity** [comment]
@cartpauj I'll publish a video tomorrow for Blair where I show what the new options look like.

File: admin/functions-admin.php [hunk 1]
```diff
@@ -48,7 +48,9 @@ function members_admin_register_scripts() {
 		'label_grant_cap'  => esc_html__( 'Grant %s capability', 'members' ),
 		'label_deny_cap'   => esc_html__( 'Deny %s capability',  'members' ),
 		'ays_delete_role'  => esc_html__( 'Are you sure you want to delete this role? This is a permanent action and cannot be undone.', 'members' ),
-		'hidden_caps'      => members_get_hidden_caps()
+        'check_all'        => esc_html__( 'Check All', 'members' ),
```
Comment (cartpauj) [inline admin/functions-admin.php:51]: @ThemeGravity I think your editor is using spaces instead of tabs here.

File: admin/tmpl/cap-section.php [hunk 2]
```diff
@@ -20,6 +20,17 @@
 				<th class="column-grant"><?php esc_html_e( 'Grant', 'members' ); ?></th>
 				<th class="column-deny"><?php esc_html_e( 'Deny', 'members' ); ?></th>
 			</tr>
+        <tr>
```
Comment (cartpauj) [inline admin/tmpl/cap-section.php:23]: @ThemeGravity same here, spaces instead of tabs.

File: css/admin.css [hunk 3]
```diff
@@ -742,6 +742,19 @@
 	border-left: 1px solid #e5e5e5;
 }
 
+.members-roles-select th, .members-roles-select td {
+    width: 190px;
+    overflow: hidden;
```
Comment (cartpauj) [inline css/admin.css:747]: @ThemeGravity more spaces.

File: js/edit-role.js [hunk 4]
```diff
@@ -266,6 +266,44 @@ jQuery( document ).ready( function() {
...
+            _.each( allDenyCheckboxes, function( checkbox ) {
+                checkbox.checked = isChecked;
+                members_check_uncheck( checkbox );
+            });
+        }
+
+        if (isChecked) {
+            $this.next('label').text(members_i18n.uncheck_all);
+            if (isGrantCheckbox) {
+                denyCheckbox.prop('checked', false).next('label').text(members_i18n.check_all);
```
Comment (cartpauj) [inline js/edit-role.js:295]: @ThemeGravity this new function is all spaces too.

**Review by cartpauj — CHANGES_REQUESTED** [review]
@ThemeGravity just replace the spaces with tabs in the various files.

I have not functionally tested this yet, just a code review.

I will update the review later after testing.

**Comment by ThemeGravity** [comment]
@cartpauj Sorry. Yeah, I have spaces in PHPStorm. I hope I have replaced all spaces with tabs.

**Comment by skorasaurus** [comment]
Thank you for your work on this. :)

**Pull request merged**